src/hooks/transactions/usePayload.ts

```diff
@@ -38,27 +39,32 @@ export const usePayload = ({ call, weight }: Input) => {
     }
   } = useSourceTarget();
   const { dispatchTransaction } = useUpdateTransactionContext();
+  const { receiverAddress } = useTransactionContext();
   const { account } = useAccountContext();
 
-  const payloadCallback = useCallback(() => {
-    if (!account || !call || !weight) {
-      return null;
-    }
+  const payloadCallback = useCallback(
+    ({ call, weight }: Input) => {
+      if (!account || !call || !weight || !receiverAddress) {
+        return null;
+      }
+
+      const payload = {
+        call: compactAddLength(call!),
+        origin: {
+          SourceAccount: account!.addressRaw
+        },
+        spec_version: targetApi.consts.system.version.specVersion.toNumber(),
+        weight
+      };
 
-    const payload = {
-      call: compactAddLength(call!),
-      origin: {
-        SourceAccount: account!.addressRaw
-      },
-      spec_version: targetApi.consts.system.version.specVersion.toNumber(),
-      weight
-    };
-    // @ts-ignore
-    const payloadType = createType(sourceChain, 'OutboundPayload', payload);
-    logger.info(`OutboundPayload: ${JSON.stringify(payload)}`);
-    logger.info(`OutboundPayload.toHex(): ${payloadType.toHex()}`);
-    return payload;
-  }, [account, call, createType, sourceChain, targetApi, weight]);
+      // @ts-ignore
+      const payloadType = createType(sourceChain, 'OutboundPayload', payload);
+      logger.info(`OutboundPayload: ${JSON.stringify(payload)}`);
```

Comment:
I would conver these to debug in order to avoid having them in production

Reply:
Why would that be an issue? How do we distinguish between "production" and "development"?